operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/updatermatcher/GenericResourceUpdaterMatcher.java

```diff
@@ -2,48 +2,40 @@
 
 import java.util.Map;
 
-import io.fabric8.kubernetes.api.model.*;
-import io.fabric8.kubernetes.api.model.discovery.v1.EndpointSlice;
-import io.fabric8.kubernetes.api.model.rbac.ClusterRole;
-import io.fabric8.kubernetes.api.model.rbac.ClusterRoleBinding;
-import io.fabric8.kubernetes.api.model.rbac.Role;
-import io.fabric8.kubernetes.api.model.rbac.RoleBinding;
-import io.javaoperatorsdk.operator.ReconcilerUtils;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.client.utils.KubernetesSerialization;
 import io.javaoperatorsdk.operator.api.reconciler.Context;
 import io.javaoperatorsdk.operator.processing.dependent.kubernetes.GenericKubernetesResourceMatcher;
 import io.javaoperatorsdk.operator.processing.dependent.kubernetes.ResourceUpdaterMatcher;
 
 public class GenericResourceUpdaterMatcher<R extends HasMetadata> implements
     ResourceUpdaterMatcher<R> {
 
+  private static final String METADATA = "metadata";
   private static final ResourceUpdaterMatcher<?> INSTANCE = new GenericResourceUpdaterMatcher<>();
 
-  @SuppressWarnings("rawtypes")
-  private static final Map<Class, ResourceUpdaterMatcher> processors = Map.of(
-      Secret.class, new SecretResourceUpdaterMatcher(),
-      ConfigMap.class, new ConfigMapResourceUpdaterMatcher(),
-      ServiceAccount.class, new ServiceAccountResourceUpdaterMatcher(),
-      Role.class, new RoleResourceUpdaterMatcher(),
-      ClusterRole.class, new ClusterRoleResourceUpdaterMatcher(),
-      RoleBinding.class, new RoleBindingResourceUpdaterMatcher(),
-      ClusterRoleBinding.class, new ClusterRoleBindingResourceUpdaterMatcher(),
-      Endpoints.class, new EndpointsResourceUpdaterMatcher(),
-      EndpointSlice.class, new EndpointSliceResourceUpdateMatcher());
-
   protected GenericResourceUpdaterMatcher() {}
 
   @SuppressWarnings("unchecked")
   public static <R extends HasMetadata> ResourceUpdaterMatcher<R> updaterMatcherFor(
       Class<R> resourceType) {
-    final var processor = processors.get(resourceType);
-    return processor != null ? processor : (ResourceUpdaterMatcher<R>) INSTANCE;
+    return (ResourceUpdaterMatcher<R>) INSTANCE;
   }
 
+  @SuppressWarnings("unchecked")
+  @Override
   public R updateResource(R actual, R desired, Context<?> context) {
-    var clonedActual = context.getControllerConfiguration().getConfigurationService()
-        .getResourceCloner().clone(actual);
+    KubernetesSerialization kubernetesSerialization =
+        context.getClient().getKubernetesSerialization();
+    Map<String, Object> actualMap = kubernetesSerialization.convertValue(actual, Map.class);
```

Comment:
Could we add unit test for this please?

Reply:
All lines are already covered by GenericResourceUpdaterMatcherTest - are there some specific cases you are concerned about?

Reply:
I meant also for the speacial cases, tole RoleBinging, ServiceAccount etc. At least for some of those which were deleted.

Reply:
> At least for some of those which were deleted.

Secret is covered by the test.  I'll add something for ServiceAccount.